### PR TITLE
[#588] Application Form: Remove "Learning period since your 1st workshop"

### LINKED ISSUE #588: Application Form: Remove "Learning period since your 1st workshop"
In 2016, the field "For how many months have you actively been learning after your first workshop?" was confusing for students who didn't attend a 'benchmark' workshop. It also didn't help us a lot during the selection process.

**Suggested solution:**  
 - Remove the field "For how many months have you actively been learning after your first workshop?".
 - Rename the field "Please summarize what you have been doing to learn programming since your first workshop"  to "Please summarize what you have been doing to learn programming".

### PR DESCRIPTION
**Pull request by bitboxer**
closes #588

### CODE REVIEW

File: db/migrate/20170110102731_remove_application_learning_period_from_users.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class RemoveApplicationLearningPeriodFromUsers < ActiveRecord::Migration[5.0]
+  def change
+    remove_column :users, :application_learning_period
```
Comment (carpodaster) [inline db/migrate/20170110102731_remove_application_learning_period_from_users.rb]: Could you add the `:text` type so the migration becomes reversible (at least on the schema-level) please?

File: app/views/application_drafts/_student_fields.html.slim [hunk 2]
```diff
@@ -35,12 +35,10 @@
       = student_field_error_hint(index, application_draft, :application_coding_level)
   = s.input :application_community_engagement, label: 'Tell us about your community involvement.', input_html: { disabled: !may_edit?(s.object) }, wrapper_html: { class: student_field_error_class(index, application_draft, :application_community_engagement) }, hint: (student_field_error_hint(index, application_draft, :application_community_engagement) || 'When did you attend your first Rails Girls, Railsbridge, Black Girls Code, PyLadies, or similar workshop? Have you been involved in organization/coaching/study groups since then? Tell us all about it!')
   .form-inline
-    = s.input :application_learning_period, :collection => User::MONTHS_LEARNING, include_blank: true, label: 'For how many months have you actively been learning after your first workshop?', disabled: !may_edit?(s.object), wrapper_html: { class: student_field_error_class(index, application_draft, :application_learning_period)}, hint: student_field_error_hint(index, application_draft, :application_learning_period)
```
Comment (carpodaster) [inline app/views/application_drafts/_student_fields.html.slim]: I think the `User::MONTHS_LEARNING` is no longer needed now. Could you verify that in your branch and remove it from the `User` model as well?

Comment (bitboxer) [inline app/views/application_drafts/_student_fields.html.slim]: It is still used for ```application_language_learning_period```

File: db/migrate/20170110102731_remove_application_learning_period_from_users.rb (hunk 1 above)
Comment (bitboxer) [inline db/migrate/20170110102731_remove_application_learning_period_from_users.rb]: Fixed.

**Comment by bitboxer** [comment]
Okay, fixed the migration and rebased the PR to the current master.

File: app/views/application_drafts/_student_fields.html.slim (hunk 2 above)
Comment (carpodaster) [inline app/views/application_drafts/_student_fields.html.slim]: Ok, thanks for checking

**Comment by carpodaster** [comment]
Thanks @bitboxer :confetti_ball:

**Pull request merged**